ida_gef.py

```diff
@@ -202,17 +202,31 @@ def Sync(self, pc, bps):
         if _current_instruction > 0:
             idc.SetColor(_current_instruction, CIC_ITEM, _current_instruction_color)
 
+        base_addr = idaapi.get_imagebase()
+        pc = base_addr + int(offset, 16)
         _current_instruction = long(pc)
         _current_instruction_color = GetColor(_current_instruction, CIC_ITEM)
         idc.SetColor(_current_instruction, CIC_ITEM, 0x00ff00)
-
-        for bp in bps:
-            if bp not in _breakpoints:
-                idc.AddBpt(bp)
-                _breakpoints.add(bp)
-
-        _new = [ idc.GetBptEA(n) for n in range(idc.GetBptQty()) ]
-        return _new
+        print("PC @ " + hex(_current_instruction).strip('L'))
+        time.sleep(0.1)    # slow it down to prevent IDA crash
```

Comment:
You can post it to the ida main thread to prevent race conditions.

```python
def f():
    "add bp, etc"

idaapi.execute_sync(f, idaapi.MFF_WRITE)
```